scripts/lib/CIME/case.py

```diff
@@ -462,6 +462,11 @@ def _set_compset(self, compset_name, files):
 
         return None, science_support
 
+    def get_primary_component(self):
+        if self._primary_component is None:
+            self._primary_component = self._find_primary_component()
```

Comment:
Please remove 

```
         # This needs to be called after self.set_comp_classes, which is called
         # from self._get_component_config_data
         self._primary_component = self._find_primary_component()
```
... from configure method.

In _set_info_from_primary_component, change
```
component = self._primary_component
```

to 
```
component = self.get_primary_component()
```